storage-account.tf

```diff
@@ -1,5 +1,94 @@
 // Shared and specialised Storage Accounts
 
+locals {
+  mgmt_network_name    = "core-cftptl-intsvc-vnet"
+  mgmt_network_rg_name = "aks-infra-cftptl-intsvc-rg"
+
+  sa_aat_subnets = [
+    data.azurerm_subnet.jenkins_subnet.id,
+    data.azurerm_subnet.aks-00-mgmt.id,
+    data.azurerm_subnet.aks-01-mgmt.id,
+    data.azurerm_subnet.aks-00-infra.id,
+    data.azurerm_subnet.aks-01-infra.id,
+    data.azurerm_subnet.aks-00-preview.id,
+  data.azurerm_subnet.aks-01-preview.id]
+
+  sa_other_subnets = [
+    data.azurerm_subnet.jenkins_subnet.id,
+    data.azurerm_subnet.aks-00-mgmt.id,
+    data.azurerm_subnet.aks-01-mgmt.id,
+    data.azurerm_subnet.aks-00-infra.id,
+  data.azurerm_subnet.aks-01-infra.id]
+
+  sa_subnets = split(",", var.env == "aat" ? join(",", local.sa_aat_subnets) : join(",", local.sa_other_subnets))
```

Comment:
I'm a bit confused on what you're trying to do here, this is just an array as far as i can tell.

I would recommend you create a list of default subnets, and a list of additional subnets that are added in aat and then concat them,

example using dummy data:

```
$ cat locals.tf
locals {
  foo = ["hello", "hello2"]
  bar = ["hello3", "hello4"]
}

$ terraform console
> local.bar
[
  "hello3",
  "hello4",
]
> concat(local.foo, local.bar)
[
  "hello",
  "hello2",
  "hello3",
  "hello4",
]
```